code/modules/spells/targeted/street_alchemy.dm

```diff
@@ -0,0 +1,140 @@
+/spell/targeted/alchemy
+	name = "Street Alchemy"
+	desc = "Gather all reagents contained in a target into an elixir. The elixir may be consumed."
+	abbreviation = "SA"
+	user_type = USER_TYPE_WIZARD
+	specialization = SSUTILITY
+	school = "transmutation"
+	charge_max = 150
+	cooldown_min = 15
+	invocation_type = SpI_NONE
+	range = 10 //If you can see it, you can steal it
+	max_targets = 1
+	spell_flags = WAIT_FOR_CLICK | NEEDSHUMAN | INCLUDEUSER
+	hud_state = "wiz_stAlch"
+	price = 0.5 * Sp_BASE_PRICE
+	spell_levels = list(Sp_SPEED = 0, Sp_POWER = 0, Sp_RANGE = 0)
+	level_max = list(Sp_TOTAL = 4, Sp_SPEED = 3, Sp_POWER = 1, Sp_RANGE = 1)
+
+/spell/targeted/alchemy/apply_upgrade(upgrade_type)
+	switch(upgrade_type)
+		if(Sp_SPEED)
+			return quicken_spell()
+		if(Sp_POWER)
+			spell_levels[Sp_POWER]++
+			name = "Permeating " + name
+			return "Thrown elixirs now transfer reagents."
+		if(Sp_RANGE)
+			spell_levels[Sp_RANGE]++
+			return "You now pilfer in an area around the target."
+
+/spell/targeted/alchemy/get_upgrade_info(upgrade_type)
+	switch(upgrade_type)
+		if(Sp_SPEED)
+			return "Reduce this spell's cooldown."
+		if(Sp_POWER)
+			return "Thrown elixirs transfer their contents into living targets."
+		if(Sp_RANGE)
+			return "Pilfers reagents in an area around the target."
+
+/spell/targeted/alchemy/cast(list/targets, mob/user)
+	for(var/target in targets)
+		if(spell_levels[Sp_RANGE])
+			aoeAlchemy(target, user)
+			return
+		singleAlchemy(target, user)
+
+/spell/targeted/alchemy/proc/singleAlchemy(target, mob/user)
+	var/obj/item/weapon/reagent_containers/pill/streetAlchElixir/A = null
+	if(spell_levels[Sp_POWER])
+		A = new /obj/item/weapon/reagent_containers/pill/streetAlchElixir/hypoElixir(src)
+	else
+		A = new /obj/item/weapon/reagent_containers/pill/streetAlchElixir(src)
+	if(user.find_empty_hand_index())
+		user.put_in_hands(A)
+	if(iscarbon(target))
+		mobAlchemy(target, user, A)
+	if((istype(target, /obj/structure)) || (istype(target, /obj/item/weapon/storage)) || (istype(target, /obj/item/weapon/reagent_containers)) || (istype(target, /obj/machinery)))
+		itemAlchemy(target, user, A)
+	if(A.is_empty())
+		to_chat(user, "You fail to perform alchemy.")
+		user.drop_item(A, force_drop = 1)
+		qdel(A)
+		return
+	to_chat(user, "You perform alchemy.")
```

Comment:
most spells don't give you confirmation feedback, probably the item appearing in your hand is confirmation enough

Reply:
> > take one half price spell
> > don't even have to upgrade it for a mere 10 points you can
> > one click to ruin the chemfridge (don't need aoe for this at all)
> > one click to ruin all the food inside the botany fridge
> > one click to deplete every bloodpack from the blood fridge
> 
> I also suspect "get chems out of things you aren't supposed to be able to" will have UNINTENDED CONSEQUENCES

I tested it on everything I could think of and the worst offender was welder tanks. A 1000u welder fuel pill sounds scary but you can't remove anything from it and it's no react. It also takes a while to actually kill anyone with the power upgrade even with that much.

As for the other issues I can turn a lot of those into upgrades. I like the idea of the spell being cheap at baseline but having the real dirty gunk you can do with it cost half your spell points or more, modular too. Maybe an upgrade that only finds X number of reagent containers in lockers/storage as well so it's not just "Have upgrade, ruin whole fridge". I'd probably just want to do either the search level or that one, or make one super cheap.